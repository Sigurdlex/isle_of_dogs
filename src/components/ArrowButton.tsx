import React, { ReactNode } from 'react';
import styled from 'styled-components';

interface ArrowButtonProps {
  children: ReactNode;
  handleClick: () => void;
  color: string;
  isLeft?: boolean;
}
type ContainerProps = Pick<ArrowButtonProps, 'color' | 'isLeft'>;

export const ArrowButton = (
  { children, handleClick, color, isLeft = false }: ArrowButtonProps
) => (
  <Container
    isLeft={isLeft}
    onClick={handleClick}
    color={color}
  >
    {children}
  </Container>
);

const Container = styled.button<ContainerProps>`
  border: none;
  font-size: 1rem;
  padding: ${({ isLeft }) => (isLeft ? '.5em 1em .5em 2em' : '.5em 2em .5em 1em')};
  background: white;
  border-radius: ${({ isLeft }) => (isLeft ? '1em 0 0 1em' : '0 1em 1em 0')};
  position: relative;
  &:before {
    position: absolute;
    content: '';
    height: 100%;
    top: 0;
    ${({ isLeft }) => (isLeft ? 'left' : 'right')}: 1em;
    width: 5px;
    background: ${({ color }) => color};
  }
`;
