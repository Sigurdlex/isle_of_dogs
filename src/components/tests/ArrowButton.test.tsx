import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import { ArrowButton } from '../ArrowButton';

afterEach(cleanup);

const handleClick = jest.fn();
const CHILDREN = 'CHILDREN';

test('<ArrowButton /> left', () => {
  const { container } = render(
    <ArrowButton handleClick={handleClick} color='white' isLeft>
      {CHILDREN}
    </ArrowButton>
  );
  const element: HTMLButtonElement = container.firstChild as HTMLButtonElement;
  expect(element.textContent).toMatch(CHILDREN);
  fireEvent.click(element);
  expect(handleClick).toHaveBeenCalledTimes(1);
  expect(element).toHaveStyleRule('padding', '.5em 1em .5em 2em');
  expect(element).toHaveStyleRule('border-radius', '1em 0 0 1em');
  expect(element).toMatchSnapshot();
});

test('<ArrowButton /> right', () => {
  const { container: { firstChild: container } } = render(
    <ArrowButton handleClick={handleClick} color='white'>
      {CHILDREN}
    </ArrowButton>
  );
  expect(container).toHaveStyleRule('padding', '.5em 2em .5em 1em');
  expect(container).toHaveStyleRule('border-radius', '0 1em 1em 0');
  expect(container).toMatchSnapshot();
});
