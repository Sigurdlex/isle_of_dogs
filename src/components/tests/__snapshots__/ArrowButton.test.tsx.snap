// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrowButton /> left 1`] = `
.c0 {
  border: none;
  font-size: 1rem;
  padding: .5em 1em .5em 2em;
  background: white;
  border-radius: 1em 0 0 1em;
  position: relative;
}

.c0::before {
  position: absolute;
  content: '';
  height: 100%;
  top: 0;
  left: 1em;
  width: 5px;
  background: white;
}

<button
  class="c0"
  color="white"
>
  CHILDREN
</button>
`;

exports[`<ArrowButton /> right 1`] = `
.c0 {
  border: none;
  font-size: 1rem;
  padding: .5em 2em .5em 1em;
  background: white;
  border-radius: 0 1em 1em 0;
  position: relative;
}

.c0::before {
  position: absolute;
  content: '';
  height: 100%;
  top: 0;
  right: 1em;
  width: 5px;
  background: white;
}

<button
  class="c0"
  color="white"
>
  CHILDREN
</button>
`;
